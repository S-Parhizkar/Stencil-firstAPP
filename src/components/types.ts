export interface ITodo {
    completed: boolean,
    id: number,
    order: number,
    title: string,
    url: string
} 
