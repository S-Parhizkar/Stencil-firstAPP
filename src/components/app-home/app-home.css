.app-home {
  padding: 10px;
}


/* h1 {
  color: #fff;
  font-weight: bold;
  font-size: 20px;
  font-family: cursive;
  text-align: center;
} */

.header{
  background-image: url(/assets/todo-list.png);
  background-repeat: no-repeat;
  background-size: 66% 135%;
  background-position-x: center;
background-position-y: center;
width: 100%;
height: 100px;
text-align: center;
}
