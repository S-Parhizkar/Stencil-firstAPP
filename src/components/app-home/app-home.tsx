import { Component, h, Listen} from '@stencil/core';

@Component({
  tag: 'app-home',
  styleUrl: 'app-home.css',
  shadow: true,
})
export class AppHome {

  @Listen("myEvent", {capture: true})
  handleMyEvent(event:any){
  console.log(event);}

  render() {
    return (
      <div class="app-home">
        <p>
          Welcome to the new API. You can use this starter to build entire apps all with web components using Stencil! Check out our docs on{' '}
          <a href="https://stenciljs.com">stenciljs.com</a> to get started.
        </p>

        <stencil-route-link url="/profile/stencil">
          <button>Profile page</button>
        </stencil-route-link>
        <my-button text='hello'> <h1 slot='fromage'>Hello world</h1></my-button>
        <my-button text='fromage'> <h1>Hello world</h1></my-button>
        <my-button text='du vin'> <h1>Hello world</h1></my-button>
      </div>
    );
  }
}
