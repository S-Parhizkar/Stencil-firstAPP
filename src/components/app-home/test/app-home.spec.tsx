import { newSpecPage, SpecPage } from '@stencil/core/testing';
import { h } from '@stencil/core';
import { AppHome } from '../app-home';
import { ITodo } from '../../types';

describe('app-home', () => {
  const todo: ITodo = {
    completed: true,
    id: 2,
    order: 3,
    title: 'sport',
    url: 'https://my-list-to-do.netlify.app/',
  };

  let page: SpecPage;

  beforeEach(async () => {
    page = await newSpecPage({
      components: [AppHome],
      template: () => <app-home></app-home>,
    });
    await page.waitForChanges();

    // element = page.root as HTMLAppHomeElement;
    // await page.waitForChanges();

    // appHome = new AppHome();
  });

  it('render', () => {
    expect(page.root).toMatchSnapshot();
  });

  // it('should listen [update-todo] event when the todo updated /checkbox is completed', async () => {
  //   const todoUp = new CustomEvent<ITodo>('update-todo');
  //   // const listenerMock = jest.fn();
  //   page.body.addEventListener('update-todo', listenerMock);
  //   // (page.root as HTMLAppHomeElement).
  //   await page.waitForChanges(element);
  //   const mockTodo = { ...todo };
  //   mockTodo.completed = true

  //   expect(todoUp).toBeTruthy(mockTodo.completed)

  // });

  it('should listen [add-more] event with add-todo component when we add new todo', async () => {
    const todoUp = new CustomEvent<ITodo>('update-todo');
    const listenerMock = jest.fn();
  
    page.body.addEventListener('add-more', listenerMock);

  page.root.shadowRoot
  await page.waitForChanges(); 
    // const mockTodo = { ...todo };
    // mockTodo.title = todo.title.toUpperCase().trim();

    expect(listenerMock).toHaveReceivedEvent({detail: todoUp});
  });
});
