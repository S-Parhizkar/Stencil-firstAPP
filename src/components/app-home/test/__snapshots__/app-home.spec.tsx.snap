// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`app-home render 1`] = `
<app-home>
  <div class="header"></div>
  <hr>
  <add-todo></add-todo>
  <hr>
  <hr>
  <div id="noloader">
    <div class="rotate"></div>
    <b>
      Loading ...
    </b>
  </div>
  <list-todo></list-todo>
</app-home>
`;
